src/tess_asteroids/movingtpf.py

```diff
@@ -910,6 +917,267 @@ def create_pixel_quality(self, sat_level: float = 1e5, sat_buffer_rad: int = 1):
             )
         self.pixel_quality = np.asarray(pixel_quality)
 
+    def to_lightcurve(self, method: str = "aperture", **kwargs):
+        """
+        Extract lightcurve from the moving TPF, using either `aperture` or `prf` photometry.
+        This function creates the `self.lc` attribute, which stores the time series data.
+
+        Parameters
+        ----------
+        method : str
+            Method to extract lightcurve. One of `aperture` or `prf`.
+        kwargs : dict
+            Keyword arguments, e.g `self._aperture_photometry` takes `bad_bits`.
+
+        Returns
+        -------
+        """
+
+        # Initialise lightcurve dictionary
+        if not hasattr(self, "lc"):
+            self.lc = {}
+
+        # Get lightcurve and quality mask via aperture photometry
+        if method == "aperture":
+            flux, flux_err, bg, bg_err, col_cen, row_cen, col_cen_err, row_cen_err = (
+                self._aperture_photometry(**kwargs)
+            )
+            quality = self._create_lc_quality()
+            self.lc["aperture"] = {
+                "time": self.time,
+                "flux": flux,
+                "flux_err": flux_err,
+                "bg": bg,
+                "bg_err": bg_err,
+                "col_cen": col_cen,
+                "col_cen_err": col_cen_err,
+                "row_cen": row_cen,
+                "row_cen_err": row_cen_err,
+                "quality": quality,
+            }
+
+        elif method == "prf":
+            raise NotImplementedError("PSF photometry is not yet implemented.")
+        else:
+            raise ValueError(
+                f"Method must be one of: ['aperture', 'prf']. Not '{method}'"
+            )
+
+    def _aperture_photometry(self, bad_bits: list = [1, 3]):
```

Comment:
The bad_bits correspond to the pixel_quality flags (i.e. those from the TPF) rather than _create_lc_quality (i.e. those for the LC). I described this in the docstrings - is that ok?